cumulusci/tasks/metadata_etl/tests/test_base.py

```diff
@@ -379,3 +381,236 @@ def test_transform__xml_parse_error(self):
+
+        entity = "CustomObject"
+        tag = "customObjectAttribute"
+        value = "newAttributeValue"
+
+        ORIGINAL_XML = f"""<?xml version="1.0" encoding="UTF-8"?>
+<{entity} xmlns="http://soap.sforce.com/2006/04/metadata">
+    <name>{api_name}</name>
+    <{tag}>oldAttributeValue</{tag}>
+    <anotherTag>value</anotherTag>
+</{entity}>
+"""
+
+        EXPECTED_XML = f"""<?xml version="1.0" encoding="UTF-8"?>
+<{entity} xmlns="http://soap.sforce.com/2006/04/metadata">
+    <name>{api_name}</name>
+    <{tag}>{value}</{tag}>
+    <anotherTag>value</anotherTag>
+</{entity}>
+"""
+
+        metadata = fromstring(ORIGINAL_XML.encode("utf-8"))
+
+        expected_metadata = fromstring(EXPECTED_XML.encode("utf-8"))
+
+        task = create_task(
+            UpdateMetadataFirstChildTextTask,
+            {
+                "managed": False,
+                "namespace_inject": None,
+                "metadata_type": entity,
+                "tag": tag,
+                "value": value,
+            },
+        )
+        task.logger = mock.Mock()
+        assert tag == task.options.get("tag")
+        assert value == task.options.get("value")
+
+        actual = task._transform_entity(metadata, api_name)
+
+        assert metadata == actual
+
+        assert actual.tostring() == expected_metadata.tostring()
+
+        task.logger.info.assert_has_calls(
+            [
+                mock.call(f'Updating {entity} "{api_name}":'),
+                mock.call(f'    {tag} as "{value}"'),
+            ]
+        )
+
+    def test_transform_entity__child_not_found(self):
+        api_name = "Supercalifragilisticexpialidocious__c"
+
+        entity = "CustomObject"
+        tag = "customObjectAttribute"
+        value = "newAttributeValue"
+
+        ORIGINAL_XML = f"""<?xml version="1.0" encoding="UTF-8"?>
+<{entity} xmlns="http://soap.sforce.com/2006/04/metadata">
+    <name>{api_name}</name>
+    <anotherTag>value</anotherTag>
+</{entity}>
+"""
+
+        EXPECTED_XML = f"""<?xml version="1.0" encoding="UTF-8"?>
+<{entity} xmlns="http://soap.sforce.com/2006/04/metadata">
+    <name>{api_name}</name>
+    <anotherTag>value</anotherTag>
+    <{tag}>{value}</{tag}>
+</{entity}>
+"""
+
+        metadata = fromstring(ORIGINAL_XML.encode("utf-8"))
+
+        expected_metadata = fromstring(EXPECTED_XML.encode("utf-8"))
+
+        task = create_task(
+            UpdateMetadataFirstChildTextTask,
+            {
+                "managed": False,
+                "namespace_inject": None,
+                "metadata_type": entity,
+                "tag": tag,
+                "value": value,
+            },
+        )
+        task.logger = mock.Mock()
+        assert tag == task.options.get("tag")
+        assert value == task.options.get("value")
+
+        actual = task._transform_entity(metadata, api_name)
+
+        assert metadata == actual
+
+        assert actual.tostring() == expected_metadata.tostring()
```

Comment:
Can't you just compare to EXPECTED_XML directly without needing to fromstring/tostring it?

Reply:
No.  I tried, but there's weird differences my guess is with indentation or line endings.

Reply:
Here you go: 53c4b7d